code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -61,6 +65,11 @@
 	remove_self(removed)
 	return
 
+
+//Initial effect is called once when the reagent first starts affecting a mob.
+/datum/reagent/proc/initial_effect(var/mob/living/carbon/M, var/alien)

```

Comment:
i added this because its quite needed and a significant hole in functionality, it will be useful in future, best to keep it